executor/join.go

```diff
@@ -572,3 +639,35 @@ func (e *NestedLoopApplyExec) Next(goCtx goctx.Context) (Row, error) {
 		e.cursor = 0
 	}
 }
+
+// NextChunk implements the Executor interface.
+func (e *NestedLoopApplyExec) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	for {
+		appendSize := mathutil.Min(e.resultChunk.NumRows()-e.cursor, e.maxChunkSize)
+		if appendSize > 0 {
+			chk.Append(e.resultChunk, e.cursor, e.cursor+appendSize)
+			e.cursor += appendSize
+		}
+		if chk.NumRows() == e.maxChunkSize {
```

Comment:
this check can be put inside `if appendSize > 0 {}`, we only change the size of `chk` in there.